### PR TITLE
Merge dev soft restart

### PR DESCRIPTION
**Pull request by AdoAdoAdo**

### CODE REVIEW

File: integrationTests/consensus/testInitializer.go [hunk 1]
```diff
@@ -245,7 +245,7 @@ func createConsensusOnlyNode(
 	pubKeys []crypto.PublicKey,
 	testKeyGen crypto.KeyGenerator,
 	consensusType string,
-	epochStartRegistrationHandler mainFactory.EpochStartNotifier,
+	epochStartRegistrationHandler factory2.EpochStartNotifier,
```
Comment (iulianpascalau) [inline integrationTests/consensus/testInitializer.go:248]: might rename this or rename the "other factory" involved

File: node/export_test.go [hunk 2]
```diff
@@ -5,6 +5,12 @@ import (
...
 
+// PutMiniblockFieldsInTransaction -
 func PutMiniblockFieldsInTransaction(tx *transaction.ApiTransactionResult, miniblockMetadata *dblookupext.MiniblockMetadata) *transaction.ApiTransactionResult {
 	return putMiniblockFieldsInTransaction(tx, miniblockMetadata)
 }
+
+// PutResultsInTransaction -
+func (n *Node) PutResultsInTransaction(hash []byte, tx *transaction.ApiTransactionResult, epoch uint32) {
+	n.putResultsInTransaction(hash, tx, epoch)
+}
```
Comment (iulianpascalau) [inline node/export_test.go:16]: Please add empty line

File: node/node.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go/dataRetriever"
 	"github.com/ElrondNetwork/elrond-go/debug"
 	"github.com/ElrondNetwork/elrond-go/facade"
-	mainFactory "github.com/ElrondNetwork/elrond-go/factory"
+	factory2 "github.com/ElrondNetwork/elrond-go/factory"
```
Comment (iulianpascalau) [inline node/node.go:25]: Please rename this or the other factory involved

File: integrationTests/consensus/testInitializer.go (hunk 1 above)
Comment (AdoAdoAdo) [reply integrationTests/consensus/testInitializer.go:248]: done

File: node/export_test.go (hunk 2 above)
Comment (AdoAdoAdo) [reply node/export_test.go:16]: done

File: node/node.go (hunk 3 above)
Comment (AdoAdoAdo) [reply node/node.go:25]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]
Reviewed commits: a11c7a0, 1bee10a, aa024b0.

**Pull request merged**